### ISSUE TITLE
Quoting messages with mentions has regressed

### ISSUE THREAD

**Issue by lukebarnard1** [vector-im/riot-web#4756]
Clicking quote (or copy-pasting the same message) throws a 
```
Uncaught Error: Unknown DraftEntity Key.
    at invariant (http://localhost:8080/bundles/_dev_/bundle.js:137598:16)
    at ContentState.getEntity (http://localhost:8080/bundles/_dev_/bundle.js:223323:51)
    at http://localhost:8080/bundles/_dev_/bundle.js:219961:60
    at http://localhost:8080/bundles/_dev_/bundle.js:221080:12
    at http://localhost:8080/bundles/_dev_/bundle.js:170282:32
    at List.__iterate (http://localhost:8080/bundles/_dev_/bundle.js:168065:14)
    at List.reduce (http://localhost:8080/bundles/_dev_/bundle.js:170277:13)
    at findRangesImmutable (http://localhost:8080/bundles/_dev_/bundle.js:221078:13)
    at ContentBlock.findEntityRanges (http://localhost:8080/bundles/_dev_/bundle.js:221030:6)
    at MessageComposerInput.findLinkEntities (http://localhost:8080/bundles/_dev_/bundle.js:219959:27)
    at http://localhost:8080/bundles/_dev_/bundle.js:224309:8
    at Array.forEach (native)
    at CompositeDraftDecorator.getDecorations (http://localhost:8080/bundles/_dev_/bundle.js:224297:23)
    at Object.generate (http://localhost:8080/bundles/_dev_/bundle.js:223118:46)
    at http://localhost:8080/bundles/_dev_/bundle.js:222870:23
    at http://localhost:8080/bundles/_dev_/bundle.js:168875:47
    at http://localhost:8080/bundles/_dev_/bundle.js:168945:19
    at http://localhost:8080/bundles/_dev_/bundle.js:168691:36
    at http://localhost:8080/bundles/_dev_/bundle.js:168558:44
    at List.__iterate (http://localhost:8080/bundles/_dev_/bundle.js:168065:14)
    at OrderedMap.__iterate (http://localhost:8080/bundles/_dev_/bundle.js:168557:26)
    at ToKeyedSequence.__iterate (http://localhost:8080/bundles/_dev_/bundle.js:168689:26)
    at KeyedIterable.filterSequence.__iterateUncached (http://localhost:8080/bundles/_dev_/bundle.js:168942:17)
    at seqIterate (http://localhost:8080/bundles/_dev_/bundle.js:166477:17)
    at KeyedIterable.Seq.__iterate (http://localhost:8080/bundles/_dev_/bundle.js:166147:15)
    at KeyedIterable.mappedSequence.__iterateUncached (http://localhost:8080/bundles/_dev_/bundle.js:168874:24)
    at seqIterate (http://localhost:8080/bundles/_dev_/bundle.js:166477:17)
    at KeyedIterable.Seq.__iterate (http://localhost:8080/bundles/_dev_/bundle.js:166147:15)
```

presumably now because entities are no longer global, so anything assuming they are is now broken. This only happens when a message that includes a mention is quoted.

**Comment by lukebarnard1** [comment]
`at MessageComposerInput.findLinkEntities` contains the call to `getEntity` that causes this error to be thrown.

**Comment by lampholder** [comment]
P1 reflects need to get quoting not erroring, however that might be implemented.

@lukebarnard1 - is our decision to simplify for now and only quote the plaintext of a message in response to this issue, or was this issue introduced independently by the bump to the new draft.js version?

**Comment by lukebarnard1** [comment]
The simplification is https://github.com/matrix-org/matrix-react-sdk/issues/1279 although this has introduced a slight issue whereby we blindly insert the markdown into the RTE causing https://github.com/vector-im/riot-web/issues/4723#issuecomment-321193742

**Comment by lukebarnard1** [comment]
We've now merged #4773, which uses the `innerText` of the EventTile, effectively doing a copy-paste. This doesn't preserve newlines.

We should just design quoting single messages and then do it.

**Comment by ara4n** [comment]
it's now fixed properly fwiw in https://github.com/matrix-org/matrix-react-sdk/pull/1890
